src/udev/udev-node.c

```diff
@@ -157,15 +156,18 @@ static int link_find_prioritized(sd_device *dev, bool add, const char *stackdir,
                 if (streq(de->d_name, id))
                         continue;
 
-                path = path_join(stackdir, de->d_name);
-                if (!path)
-                        return -ENOMEM;
-
-                if (readlink_malloc(path, &buf) >= 0) {
+                if (de->d_type == DT_LNK) {
```

Comment:
hmm, readdir() might actually return DT_UNKNOWN in d_type. At the very least, use dirent_ensure_type() to make sure we can rely on it.

Reply:
dirent_ensure_type() is called in FOREACH_DIRENT_ALL().